html-aam/accname-computation-by-element/tr-td-th.html

```diff
@@ -0,0 +1,48 @@
+<!doctype html>
+<html>
+
+<head>
+  <meta charset="utf-8">
+  <title>Accname computation tests for &lt;tr&gt;</title>
+  <script src="/resources/testharness.js"></script>
+  <script src="/resources/testharnessreport.js"></script>
+  <script src="/resources/testdriver.js"></script>
+  <script src="/resources/testdriver-vendor.js"></script>
+  <script src="/resources/testdriver-actions.js"></script>
+  <script src="/wai-aria/scripts/aria-utils.js"></script>
+  <script src="/html-aam/scripts/html-aam-accname-utils.js"></script>
+</head>
+
+<body>
+  <script>
+    const testConfig = {
+      urlSpec: 'https://www.w3.org/TR/html-aam-1.0/#tr-td-th-elements-accessible-name-computation',
+      elements: {
+        'tr': { parents: ['table', 'tbody'] },
+        'th': { parents: ['table', 'thead', 'tr'] },
+        'td': { parents: ['table', 'tbody', 'tr'] },
+
+        /*
+          PR #44965 TODO:
+          How about thead, tbody, and tfoot, which support name from author?
+        */
+
+        'thead': { parents: ['table'] },
+        'tbody': { parents: ['table'] },
+        'tfoot': { parents: ['table'] },
```

Comment:
I'm not certain these tests would be valid w/o the relevant descendants, would they? IIRC, engines invalidate some containers if other required context is missing or invalid, so a `table` that contained no cell children would not be a table, and therefore your downstream `thead` expectations may be invalid.

Note: I'm reading this first one as testing `<table><thead></thead></table>` so if I'm missing some cell additions later, please correct.

Reply:
Oof, I didn’t even consider that. I’ve filled out the table structures for [all the `table`-related subtests](https://github.com/web-platform-tests/wpt/pull/44965/commits/80544531d21b0b90ef68b98610224b574b2cd2a6).

I did a quick once-over for the other similarly non-trivial markup structures — `area`, `ruby`, `summary`, etc. — and I believe those are all sufficiently complete.

Reply:
Now that’s interesting. After implementing your suggestion here, [Chrome has 2 new failures and Firefox has 1](https://wpt.fyi/results/html-aam/accname-computation-by-element/tr-td-th.html?label=pr_head&max-count=1&pr=44965).

With the previous (incomplete) subtests, [everything passed in flying colors](https://wpt.fyi/results/html-aam/accname-computation-by-element/tr-td-th.html?sha=84f1c0d&label=pr_head).

With the `table` elements now fully fleshed out, Chrome is happy to let both `th` and `td` get their names from their contents, which [the HTML-AAM spec doesn’t seem to permit](https://www.w3.org/TR/html-aam-1.0/#tr-td-th-elements-accessible-name-computation). This is also true for Firefox, but only for `th`.

Reply:
I would then recommend marking this file (or just those tests in a new file) as tentative. Then spawn a new ARIA issue for the WG to determine what the "correct" behavior should be.

Reply:
Thanks, @cookiecrook. I’ve [marked that file as tentative](https://github.com/web-platform-tests/wpt/pull/44965/commits/ea6ca4cee9da1b02e5547c2ea1c988a135ee7729).

I also found that issue appears to exist already in w3c/html-aam#543, so I left [a comment](https://github.com/w3c/html-aam/issues/543#issuecomment-2415714902) to mention this test.